### PR TITLE
Fixed #28148 -- Documented ImageField behaviour change

### PR DESCRIPTION
**Pull request by xrmx**
Since 1.11 images names are validated.

### CODE REVIEW

File: docs/topics/testing/tools.txt [hunk 1]
```diff
@@ -235,7 +235,12 @@ Use the ``django.test.Client`` class to make requests.
         file-processing code expects.)
 
         You may also provide any file-like object (e.g., :class:`~io.StringIO` or
-        :class:`~io.BytesIO`) as a file handle.
+        :class:`~io.BytesIO`) as a file handle but in order to satisfy
+        :data:`~django.core.validators.validate_image_file_extension` validator
```
Comment (timgraham) [inline docs/topics/testing/tools.txt]: In the midst of discussion of general file upload, suddenly there's a mention of something that's  specific to `ImageField`. I think the phrasing could be improved.

Comment (xrmx) [inline docs/topics/testing/tools.txt]: Yeah, will add a bit more context

**Comment by xrmx** [comment]
I added it at first but after grepping ImageField i found it's already mentioned there https://docs.djangoproject.com/en/1.11/releases/1.11/#models

So it looked like a duplication

**Comment by xrmx** [comment]
Updated PR to add more context about ImageField.

**Comment by timgraham** [comment]
The mention in the release notes is under "Minor features" not "Backwards incompatible changes". A note in "Backwards incompatible changes" could also include a mention of this issue.

**Comment by xrmx** [comment]
Rebased and added this issue in the "Backwards incompatible changes" section too.

**Pull request merged**